### PR TITLE
IP-361 - Migrate Cancer IR from 4.2.0 to 4.0.0

### PR DESCRIPTION
**Pull request by greglever**
[IP-361 - Migrate Cancer IR from 4.2.0 to 4.0.0](https://jira.extge.co.uk/browse/IP-361)
==================================

Summary of Changes
=================
* Add `MigrateReports420SNAPSHOTTo400.migrate_cancer_interpretation_request`
* Add `MigrationParticipants104SNAPSHOTTo100.migrate_cancer_participant`
* Add tests

Tests and Build
================
- [x]  All tests passing locally
```
(.env) $ python -m unittest discover
..................................................................
----------------------------------------------------------------------
Ran 66 tests in 3.047s

OK
(.env) $ 
```

### CODE REVIEW

File: protocols/migration/participants.py [hunk 1]
```diff
@@ -2,7 +2,87 @@
...
+            jsonDict=tumour_sample.toJsonDict()
+        )
+
+        migrated_tumour_sample.tumourId = None
+        if tumour_sample.tumourId is not None:
+            migrated_tumour_sample.tumourId = int(tumour_sample.tumourId)
+
+        migrated_tumour_sample.tumourType = None
+        if tumour_sample.diseaseType is not None:
+            migrated_tumour_sample.tumourType = int(tumour_sample.diseaseType)
```
Comment (priesgo) [inline protocols/migration/participants.py]: `diseaseType` and `tumourType` are enums, int() casting should fail and does not make any sense.

File: protocols/migration/participants.py [hunk 2]
```diff
@@ -2,7 +2,87 @@
...
+
+        migrated_tumour_sample.tumourSubType = None
+        if tumour_sample.diseaseSubType is not None:
+            migrated_tumour_sample.tumourSubType = int(tumour_sample.diseaseSubType)
```
Comment (priesgo) [inline protocols/migration/participants.py]: `diseaseSubtype` and `tumourSubtype` is a string

File: protocols/migration/participants.py [hunk 3]
```diff
@@ -2,7 +2,87 @@
...
+
+        migrated_tumour_sample.phase = None
+        if tumour_sample.tumourType is not None:
+            migrated_tumour_sample.phase = int(tumour_sample.tumourType)
```
Comment (priesgo) [inline protocols/migration/participants.py]: string?

**Comment by greglever** [comment]
Thanks for the review @priesgo 
I've added in your suggestions and put in some explicit tests, so it's ready for review again

**Pull request merged**